alternator/streams.cc

```diff
@@ -479,50 +476,135 @@ future<executor::request_return_type> executor::describe_stream(client_state& cl
     // or on expired...
     // TODO: maybe add secondary index to topology table to enable this?
     return _sdks.cdc_get_versioned_streams({ tm.count_normal_token_owners() }).then([this, &db, schema, shard_start, limit, ret = std::move(ret), stream_desc = std::move(stream_desc)](std::map<db_clock::time_point, cdc::streams_version> topologies) mutable {
-        auto i = topologies.begin();
+
+        // filter out cdc generations older than the table or now() - dynamo_streams_max_window (24h)
```

Comment:
Why is this needed to group streams by vnodes?

Reply:
Ok. Nevermind. I think I got it.

Reply:
It is technically not, but it filters out useless data, so helps further reduce shard bloat. Could technically have been a separate patch, yes.